shaders/src/light_punctual.fs

```diff
@@ -172,9 +172,10 @@ void evaluatePunctualLights(const PixelParams pixel, inout vec3 color) {
         if (light.NoL > 0.0){
             if (light.castsShadows) {
 #if defined(HAS_VSM)
-                highp float fragDepth = length(vertex_worldPosition - light.worldPosition);
-                visibility = shadowVsm(light_shadowMap, light.shadowLayer,
-                    getSpotLightSpacePosition(light.shadowIndex), fragDepth);
+                highp vec3 lightSpacePosition = getSpotLightSpacePosition(light.shadowIndex);
+                // Flip the sign of the Z coordinate; -Z is forward in light space.
+                lightSpacePosition.z *= -1.0;
```

Comment:
You should be able to include the negation in the matrix, right? (since it's already a weird matrix anyways) -- this way you wouldn't need a #if HAS_VSM case here.